target/linux/generic/hack-4.14/401-mtd-enforce-access-mode.patch

```diff
@@ -0,0 +1,60 @@
+--- a/drivers/mtd/mtdchar.c
++++ b/drivers/mtd/mtdchar.c
+@@ -595,7 +595,10 @@
+ 		/* Sanitize user input */
+ 		p.devname[BLKPG_DEVNAMELTH - 1] = '\0';
+ 
+-		return mtd_add_partition(mtd, p.devname, p.start, p.length);
++		/* No mtd flags masking required */
++		uint32_t mask_flags = 0;
```

Comment:
Did you try compiling this? That line will cause a compiler warning.

Reply:
You are indeed right:
```
drivers/mtd/mtdchar.c: In function 'mtdchar_blkpg_ioctl':
drivers/mtd/mtdchar.c:599:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   uint32_t mask_flags = 0;
   ^~~~~~~~
```
I have implemented it using C99 and missed the warning - this is now fixed.